diff --git a/src/pthread/once.cpp b/src/pthread/once.cpp
index 5292adf..9195052 100644
--- a/src/pthread/once.cpp
+++ b/src/pthread/once.cpp
@@ -5,7 +5,7 @@
 
 #include <boost/thread/detail/config.hpp>
 #ifdef BOOST_THREAD_ONCE_ATOMIC
-#include "./once_atomic.cpp"
+#include "./once_atomic.ipp"
 #else
 #define __STDC_CONSTANT_MACROS
 #include <boost/thread/pthread/pthread_mutex_scoped_lock.hpp>
diff --git a/src/win32/thread.cpp b/src/win32/thread.cpp
index 972b731..97ca4d8 100644
--- a/src/win32/thread.cpp
+++ b/src/win32/thread.cpp
@@ -5,6 +5,10 @@
 // (C) Copyright 2007 David Deakins
 // (C) Copyright 2011-2013 Vicente J. Botet Escriba
 
+#include <boost/config.hpp>
+
+#if BOOST_WINDOWS
+
 #ifndef _WIN32_WINNT
 #define _WIN32_WINNT 0x400
 #endif
@@ -1046,3 +1050,4 @@ namespace boost
 //}
 }
 
+#endif
